src/timelines.js

```diff
@@ -83,7 +83,8 @@ export default Kapsule({
                   label: rawData[i].data[j].label,
                   timeRange: rawData[i].data[j].data[k].timeRange.map(d => new Date(d)),
                   val: rawData[i].data[j].data[k].val,
-                  labelVal: rawData[i].data[j].data[k][rawData[i].data[j].data[k].hasOwnProperty('labelVal')?'labelVal':'val']
+                  labelVal: rawData[i].data[j].data[k][rawData[i].data[j].data[k].hasOwnProperty('labelVal')?'labelVal':'val'],
+                  data: rawData[i].data[j].data[k].data
```

Comment:
I think for this purpose it may be more convenient to just let all the other attributes of the segment object pass through instead. Something like:

```js
              const { timeRange, val, labelVal, ...other } = rawData[i].data[j].data[k];
              state.completeFlatData.push({
                  ...other,
                  group: group,
                  label: rawData[i].data[j].label,
                  timeRange: timeRange.map(d => new Date(d)),
                  val,
                  labelVal: labelVal !== undefined ? labelVal : val
                });
```

Reply:
Or alternatively, just attach the original segment object as a `data` attribute, which can be used later to surface in the interaction methods (click, hover, etc...). That is probably the cleanest:
```js
    data: rawData[i].data[j].data[k]
```

Reply:
I was thinking of the spread operator originally so I think that is good.